repose-aggregator/core/repose-core/src/main/java/org/openrepose/core/services/deploy/ArtifactManager.java

```diff
@@ -228,7 +233,11 @@ public boolean hasFilter(String filterName) {
 
     @Override
     public boolean allArtifactsLoaded() {
-        return !containerConfigurationListener.getDirWatcher().checkArtifacts();
+        // todo: Adding a processing flag minimizes the window during which this method may return a false positive,
+        // todo: however a false positive is still possible if this method is called while an ApplicationArtifactEvent
+        // todo: is in the event queue (i.e., the ArtifactDirectoryWatcher has noticed a change and emitted an event,
+        // todo: but the event has not yet been passed to this class, the ArtifactManager).
```

Comment:
Since it may not be obvious, this change fixed the `ArtifactDeploymentTest` which attempts to verify the artifacts are deployed as expected.

Note that this issue should not really affect end-users. Rather than short-circuiting because this method returns `false`, the `ReposeFilterLoader` will attempt to update its configuration including building the necessary filter contexts. If an artifact necessary to build a filter context has not been loaded, the update will fail. If this happens during startup, Repose will not be marked as ready until the update passes, which should happen at some point in the future (after the necessary artifact is loaded). So at worst, this issue may lead to some confusion during debugging.

**This is a reminder to myself to write a story to capture this issue.**